File: execution/backend/create_bucket.py
import os
from pathlib import Path

def load_env():
    p = Path("../.env")
    if not p.exists():
        p = Path(".env")
    if p.exists():
        for line in p.read_text().splitlines():
            line = line.strip()
            if line and not line.startswith("#") and "=" in line:
                k, v = line.split("=", 1)
                os.environ.setdefault(k.strip(), v.strip())
    else:
        print("Erro: .env não encontrado.")
        exit(1)

load_env()

url = os.environ.get("SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")

if not url or not key:
    print("Erro: Credenciais do Supabase ausentes.")
    exit(1)

from supabase import create_client, Client
supabase: Client = create_client(url, key)

bucket_name = "contracts"
try:
    buckets = supabase.storage.list_buckets()
    exists = any(b.name == bucket_name for b in buckets)
    if exists:
        print(f"Bucket '{bucket_name}' já existe.")
    else:
        print(f"Criando bucket '{bucket_name}' público...")
        # Nota: O python SDK oficial atual pode usar create_bucket direto
        supabase.storage.create_bucket(bucket_name, options={"public": True, "fileSizeLimit": 5242880})
        print("Bucket contracts criado e público!")
except Exception as e:
    print(f"Erro ao criar/verificar bucket: {e}")

File: execution/backend/create_bucket.js
'use strict';

const { createClient } = require('@supabase/supabase-js');
const { readFileSync } = require('fs');
const path = require('path');

// .env está em Projects/ (dois níveis acima de execution/backend/)
const envPath = path.resolve(__dirname, '../../.env');
const envVars = Object.fromEntries(
  readFileSync(envPath, 'utf8')
    .split('\n')
    .filter(l => l.includes('=') && !l.startsWith('#'))
    .map(l => { const i = l.indexOf('='); return [l.slice(0, i).trim(), l.slice(i + 1).trim()]; })
);

const supabaseUrl = envVars.SUPABASE_URL;
const supabaseServiceKey = envVars.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseServiceKey) {
  console.error('❌ SUPABASE_URL ou SUPABASE_SERVICE_ROLE_KEY não encontrados em .env');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseServiceKey, {
  auth: { persistSession: false },
});

const BUCKET_NAME = 'contracts';

async function ensureBucket() {
  console.log(`🔍 Verificando bucket "${BUCKET_NAME}"...`);

  const { data: buckets, error: listError } = await supabase.storage.listBuckets();
  if (listError) {
    console.error('❌ Erro ao listar buckets:', listError.message);
    process.exit(1);
  }

  const exists = buckets.some(b => b.name === BUCKET_NAME);

  if (exists) {
    console.log(`✅ Bucket "${BUCKET_NAME}" já existe. Nenhuma ação necessária.`);
    return;
  }

  console.log(`📦 Bucket não encontrado. Criando "${BUCKET_NAME}" como público...`);

  const { data, error: createError } = await supabase.storage.createBucket(BUCKET_NAME, {
    public: true,
    fileSizeLimit: 10485760, // 10 MB
    allowedMimeTypes: ['application/pdf', 'image/png', 'image/jpeg'],
  });

  if (createError) {
    console.error('❌ Erro ao criar bucket:', createError.message);
    process.exit(1);
  }

  console.log(`✅ Bucket "${BUCKET_NAME}" criado com sucesso!`);
  console.log(`   URL pública base: ${supabaseUrl}/storage/v1/object/public/${BUCKET_NAME}/`);
}

ensureBucket();
